csharp/RowGroupMetaData.cs

```diff
@@ -35,6 +35,68 @@ internal RowGroupMetaData(IntPtr handle)
         /// </summary>
         public long TotalByteSize => ExceptionInfo.Return<long>(_handle, RowGroupMetaData_Total_Byte_Size);
 
+        /// <summary>
+        /// Gets the columns by which the data is sorted in this row group.
+        /// </summary>
+        /// <returns>
+        /// An array of <see cref="WriterProperties.SortingColumn"/> specifying the sorting order for this row group,
+        /// or an empty array if no sorting columns are defined.
+        /// </returns>
+        public WriterProperties.SortingColumn[] SortingColumns()
+        {
+            IntPtr columnIndicesPtr = IntPtr.Zero;
+            IntPtr descendingPtr = IntPtr.Zero;
+            IntPtr nullsFirstPtr = IntPtr.Zero;
+            int numColumns = 0;
+
+            try
+            {
+                ExceptionInfo.Check(RowGroupMetaData_Sorting_Columns(
+                    _handle,
+                    ref columnIndicesPtr,
+                    ref descendingPtr,
+                    ref nullsFirstPtr,
+                    ref numColumns));
+
+                var columnIndices = new int[numColumns];
+                var isDescending = new bool[numColumns];
+                var nullsFirst = new bool[numColumns];
+
+                if (numColumns > 0)
+                {
+                    // Read column indices
+                    Marshal.Copy(columnIndicesPtr, columnIndices, 0, numColumns);
+
+                    // Read descending flags 
+                    for (var i = 0; i < numColumns; ++i)
+                    {
+                        isDescending[i] = Marshal.ReadByte(descendingPtr, i) != 0;
+                    }
+
+                    // Read nulls_first flags
+                    for (var i = 0; i < numColumns; ++i)
+                    {
+                        nullsFirst[i] = Marshal.ReadByte(nullsFirstPtr, i) != 0;
+                    }
+                }
+
+                // Create and return SortingColumn array
+                var result = new WriterProperties.SortingColumn[numColumns];
+                for (var i = 0; i < numColumns; i++)
+                {
+                    result[i] = new WriterProperties.SortingColumn(columnIndices[i], isDescending[i], nullsFirst[i]);
+                }
+                return result;
+            }
+            finally
+            {
+                if (columnIndicesPtr != IntPtr.Zero || descendingPtr != IntPtr.Zero || nullsFirstPtr != IntPtr.Zero)
```

Comment:
I think that the null checks are not needed, as deleting a null pointer has no effect.